### ISSUE TITLE
Isssue in loading of hierarchical data

### ISSUE THREAD

**Issue by TobiasBerndt** [database-rider/database-rider#554]
I have below dataset and entity.
When JUnit test loads the dataset via `@DataSet` SomeEntity with id 2 should have a reference to SomeEntity with id 1 in the parent attribute. However the parent attribute is null.

```
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "some_table")
public class SomeEntity {
   @Id
   @Column(name = "sk")
   private Long id;

   @ManyToOne
   @JoinColumn(name = "parent_sk")
   private SomeEntity parent;
}
```

```
<dataset>
   <some_table sk="1" />
   <some_table sk="2" parent_sk="1" />
</dataset>
```

**Comment by coiouhkc** [comment]
Could you please provide a full reproducer project?
I had a similar issue with hierarchical data where a `ConstraintViolationException` would be thrown during test execution due to missing parent -- what (i think) happened is that i had statement batching turned on (`batchedStatements:  true`) which would apparently lead to order of insert being different from the order specified by the dataset file.
Long story short, disabling batching (please also check your JPA/Hibernate configuration and  connection string for any additional configuration attributes, see eg. https://www.baeldung.com/jpa-hibernate-batch-insert-update) helped in my case.
